operator/controllers/actionsrunner/reconciler.go

```diff
@@ -0,0 +1,298 @@
+		}
+	case err := <-err:
+		return err
+	default:
+		return nil
+	}
+}
+
+func (r *Reconciler) onPipelineAgentJobRequest(wire *Wire, ack <-chan struct{}, message facades.Message) error {
+	r.Log.Info("On PipelineAgentJobRequest")
+
+	if wire == nil {
+		return errors.New("wire == nil")
+	}
+
+	if message.Type != facades.MessageTypePipelineAgentJobRequest {
+		return fmt.Errorf("message.Type != MessageTypePipelineAgentJobRequest")
+	}
+
+	desiredActionsRunnerJob, err := toActionsRunnerJob(wire.ActionsRunner, r.Scheme)
+	if err != nil {
+		return err
+	}
+	if err := r.Patch(wire.Context, desiredActionsRunnerJob, client.Apply, patchOpts...); err != nil {
+		return err
+	}
+
+	desiredPersistentVolumeClaim, err := toPersistentVolumeClaim(wire.ActionsRunner, desiredActionsRunnerJob, r.Scheme)
+	if err != nil {
+		return err
+	}
+	if err := r.Patch(wire.Context, desiredPersistentVolumeClaim, client.Apply, patchOpts...); err != nil {
+		return err
+	}
+
+	desiredJob, err := toJob(wire.ActionsRunner, desiredActionsRunnerJob, r.Scheme)
+	if err != nil {
+		return err
+	}
+	if err := r.Patch(wire.Context, desiredJob, client.Apply, patchOpts...); err != nil {
+		return err
+	}
+
+	key := fmt.Sprintf("%s/%s", desiredActionsRunnerJob.GetNamespace(), desiredActionsRunnerJob.GetName())
+	r.waiting.Store(key, nil)
+	go r.waitForJob(wire.Context, desiredActionsRunnerJob, ack)
+
+	return nil
+}
+
+// TODO
+func (r *Reconciler) onJobCancellation(wire *Wire, ack <-chan struct{}, message facades.Message) error {
+	r.Log.Info("On JobCancellation")
+
+	if wire == nil {
+		return errors.New("wire == nil")
+	}
+
+	if message.Type != facades.MessageTypeJobCancellation {
+		return errors.New("message.Type != MessageTypeJobCancellation")
+	}
+
+	actionsRunnerJob, err := toActionsRunnerJob(wire.ActionsRunner, r.Scheme)
+	if err != nil {
+		return err
+	}
+
+	if err := r.Delete(wire.Context, actionsRunnerJob, deleteOpts...); err != nil {
+		r.Log.Error(err, "r.Delete")
+	}
+
+	<-ack
+	return nil
+}
+
+// TODO
+func (r *Reconciler) onAgentRefresh(wire *Wire, ack <-chan struct{}, message facades.Message) error {
+	r.Log.Info("On AgentRefresh")
+
+	if wire == nil {
+		return errors.New("wire == nil")
+	}
+
+	if message.Type != facades.MessageTypeAgentRefresh {
+		return errors.New("message.Type != MessageTypeAgentRefresh")
+	}
+
+	<-ack
+	return nil
+}
+
+func (r *Reconciler) waitForJob(ctx context.Context, actionsRunnerJob *inlocov1alpha1.ActionsRunnerJob, ack <-chan struct{}) {
+	objectKey := client.ObjectKey{
+		Namespace: actionsRunnerJob.GetNamespace(),
+		Name:      actionsRunnerJob.GetName(),
+	}
+
+	defer func() {
+		if ack != nil {
+			<-ack
+		}
+
+		r.waiting.Delete(objectKey.String())
+	}()
+
+	condition := func() (done bool, err error) {
+		var job batchv1.Job
+		if err := r.Get(ctx, objectKey, &job); client.IgnoreNotFound(err) != nil {
+			return false, err
+		}
+
+		active := int(job.Status.Succeeded+job.Status.Failed) < len(job.Spec.Template.Spec.Containers)
```

Comment:
`isRunning`?

`is` at the beginning sounds great in bool variables

Reply:
Actually the Job may be active but not running, for example when you just created the job but no pods have already been started.

Reply:
But I'll use the prefix `is`